packages/blade/src/components/Tabs/TabIndicator.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable consistent-return */
+import React from 'react';
+import styled from 'styled-components';
+import { useTabsContext } from './TabsContext';
+import { castWebType, makeMotionTime, useTheme } from '~utils';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+import BaseBox from '~components/Box/BaseBox';
+
+const StyledTabIndicator = styled(BaseBox)(({ theme }) => {
+  return {
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.colors.brand.primary[400],
+    },
+  };
+});
+
+const TabIndicator = ({
+  tabListContainerRef,
+}: {
+  tabListContainerRef: React.RefObject<HTMLElement | null>;
+}): React.ReactElement => {
+  const { theme } = useTheme();
+  const { selectedValue, baseId, variant } = useTabsContext();
+  const [hasMeasured, setHasMeasured] = React.useState(false);
+  const [activeElementDimensions, setActiveElementDimensions] = React.useState({
+    width: 0,
+    height: 0,
+    x: 0,
+    y: 0,
+  });
+
+  // Update the dimensions of the active element
+  const updateDimensions = React.useCallback(() => {
```

Comment:
why do we change the width and height when the tabitem is selected?

Reply:
This is the bottom blue indicator that animates from 1 tabitem to another tabitem, we need recalculate and update it's dimensions. 🤔